### PR TITLE
Handle '@' in username

### PR DESCRIPTION
**Pull request by ablyes**
handle the case where the username contains one or more '@' char (like an email address used as username).

### CODE REVIEW

**Comment by ablyes** [comment]
@googlebot I signed it!

File: coursier.bzl [hunk 1]
```diff
@@ -440,7 +440,8 @@ def remove_auth_from_url(url):
     host = url_parts[0]
     if "@" not in host:
         return url
-    userless_host = host[host.find("@") + 1:]
+    last_index=host.rfind("@", 0, None)
```
Comment (plaird) [inline coursier.bzl:443]: FYI there are some split_url() tests that have weird urls that this probably won't work for. Not sure if those are real cases somebody has, or not. 

https://github.com/bazelbuild/rules_jvm_external/blob/master/tests/unit/coursier_test.bzl#L208

Comment (ablyes) [reply coursier.bzl:443]: I think it's not realistic to have a host with @ 
My commit should fix those 2 cases:
username@whatever.com:password@host
username@whatever.com@itLikes@icanputanotheronehere@:password_with_@_why_not:host

Comment (plaird) [reply coursier.bzl:443]: I agree, I don't think having that @ char in the path, in unencoded form, is even legal. But just FYI someone thought it was a valid use case.

**Review by jin — COMMENTED** [review]
Could you add tests?

**Comment by ablyes** [comment]
Yes, I will do that. 
May be just delete the none realistic tests also.

**Review by shs96c — APPROVED** [review]
This makes sense. Thank you for the PR!

**Pull request merged**